### ISSUE TITLE
`ANN001` should maybe ignore `self`

### ISSUE THREAD

**Issue by matteosantama** [charliermarsh/ruff#635]
I notice that when I enable the `ANN001` code, I get the error

```
ANN001 Missing type annotation for `self` in method
```

I would suggest ignoring the `self` parameter as it is rather self-explanatory and rarely annotated.

**Comment by matteosantama** [comment]
It also crops up as `ANN101`,

```
ANN101 Missing type annotation for `self` in method
```

I am also unable to ignore it by including it in the `ignore = [...]` section.

**Comment by tekumara** [comment]
Do you have code snippet that shows where it fails? I can ignore `ANN101` when `self` appears on a function inside a class, eg:
```python
class MyClass:

    def hello(self) -> None:
        pass
```

**Comment by tekumara** [comment]
If `self` is an argument to function that is outside of a class, then I think it's just treated like any other function argument missing an annotation.

**Comment by charliermarsh** [comment]
Can you paste a snippet to reproduce? Is this happening for functions defined outside of classes, or functions defined within classes?

(Being unable to ignore seems like a bug.)

**Comment by charliermarsh** [comment]
Ok, #637 fixes the `--ignore` issue. I'm going to close for now as it's expected that `self` requires an annotation as long as ANN101 is enabled. (I personally would not enable that rule.)
